.github/workflows/main.yaml

```diff
@@ -0,0 +1,345 @@
+on: 
+  push: 
+    tags:
+     - '*'
+
+jobs:
+  get-artifact-list:
+    runs-on: ubuntu-latest
+    steps:
+      - name: get the list
+        run: |
+         git clone https://github.com/amshinde/kata-packaging
+         pushd kata-packaging
+         tag=`echo $GITHUB_REF | cut -d/ -f3-`
+         git checkout $tag
+         popd
+         ./kata-packaging/artifact-list.sh > artifact-list.txt
+      - name: save-artifact-list
+        uses: actions/upload-artifact@master
+        with: 
+          name: artifact-list 
+          path: artifact-list.txt
+  build-kernel:
+    runs-on: ubuntu-16.04
+    needs: get-artifact-list
+    steps:
+      - name: get-artifact-list
+        uses: actions/download-artifact@master
+        with:
+          name: artifact-list
+      - name: build-kernel
+        run: |
+         if grep -q "install_kernel" ./artifact-list/artifact-list.txt; then
+           # install kernel dependencies
+           sudo apt-get update && sudo apt install -y flex bison libelf-dev bc iptables
```

Comment:
Nit: This could be `sudo apt update` (for parity with the following `sudo apt install`).